lua/cmp/utils/window.lua

```diff
@@ -127,50 +127,30 @@ end
 
 ---Update
 window.update = function(self)
-  if self:has_scrollbar() then
-    local total = self:get_content_height()
-    local info = self:info()
-    local bar_height = math.ceil(info.height * (info.height / total))
-    local bar_offset = math.min(info.height - bar_height, math.floor(info.height * (vim.fn.getwininfo(self.win)[1].topline / total)))
-    local style1 = {}
-    style1.relative = 'editor'
-    style1.style = 'minimal'
-    style1.width = 1
-    style1.height = info.height
-    style1.row = info.row
-    style1.col = info.col + info.width - (info.has_scrollbar and 1 or 0)
-    style1.zindex = (self.style.zindex and (self.style.zindex + 1) or 1)
-    if self.swin1 and vim.api.nvim_win_is_valid(self.swin1) then
-      vim.api.nvim_win_set_config(self.swin1, style1)
+  local info = self:info().scrollbar
+  if info and info.width > 0 then
+    info.relative = 'editor'
+    info.style = 'minimal'
+    info.zindex = (self.style.zindex and (self.style.zindex + 2) or 2)
+    if self.swin and vim.api.nvim_win_is_valid(self.swin) then
+      vim.api.nvim_win_set_config(self.swin, info)
     else
-      style1.noautocmd = true
-      self.swin1 = vim.api.nvim_open_win(buffer.ensure(self.name .. 'sbuf1'), false, style1)
-      vim.api.nvim_win_set_option(self.swin1, 'winhighlight', 'EndOfBuffer:PmenuSbar,Normal:PmenuSbar,NormalNC:PmenuSbar,NormalFloat:PmenuSbar')
-    end
-    local style2 = {}
-    style2.relative = 'editor'
-    style2.style = 'minimal'
-    style2.width = 1
-    style2.height = bar_height
-    style2.row = info.row + bar_offset
-    style2.col = info.col + info.width - (info.has_scrollbar and 1 or 0)
-    style2.zindex = (self.style.zindex and (self.style.zindex + 2) or 2)
-    if self.swin2 and vim.api.nvim_win_is_valid(self.swin2) then
-      vim.api.nvim_win_set_config(self.swin2, style2)
-    else
-      style2.noautocmd = true
-      self.swin2 = vim.api.nvim_open_win(buffer.ensure(self.name .. 'sbuf2'), false, style2)
-      vim.api.nvim_win_set_option(self.swin2, 'winhighlight', 'EndOfBuffer:PmenuThumb,Normal:PmenuThumb,NormalNC:PmenuThumb,NormalFloat:PmenuThumb')
-    end
```

Comment:
Please don't remove `native pum style scrollbar area`.

Reply:
I'll add it back for when there is no border 👍

I thought it would be a good place to get some performance back (since it's one more window we don't have to open), but I can understand wanting parity with the `pum`